src/Native/Unix/System.Native/pal_io.c

```diff
@@ -1225,73 +1274,135 @@ int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destinationFd)
         ssize_t sent = sendfile(outFd, inFd, NULL, (size >= SSIZE_MAX ? SSIZE_MAX : (size_t)size));
         if (sent < 0)
         {
-            if (errno != EINVAL && errno != ENOSYS)
-            {
-                return -1;
-            }
-            else
+            if (errno == ENOSYS)
             {
-                break;
+                sendFileUnavailable = true; // Benign race.
```

Comment:
Would you mind adding some more details to this comment? Why it's considered benign?

Reply:
Is this related to the bug you mentioned in the original issue that happens when an error is thrown in the middle of a copy and we end up copying everything again?

Reply:
It's somewhat related.

The race here is benign because if two concurrent calls to `sendfile()` are made on a system where it's not enabled, both will just set the boolean flag to `true` and proceed to return the number of bytes that need to be copied; the flag is the only shared resource here, so there's no need to protect this with a more complex construct.